core/state_processor.go

```diff
@@ -402,6 +404,37 @@ func ApplyIncomingReceipt(
 	return nil
 }
 
+func applyReimbursement(config *params.ChainConfig, bc BlockChain, db *state.DB, header *block.Header) {
+	if config.IsReimbursementEpoch(header.Epoch()) {
+		parentHeader := bc.GetHeader(header.ParentHash(), header.Number().Uint64()-1)
+		reimbursement := shard.Schedule.InstanceForEpoch(header.Epoch()).Reimbursement()
+		startNumber := shard.Schedule.EpochLastBlock(config.ReimbursementEpoch.Uint64()-1) + 1
+		startHeader := bc.GetHeaderByNumber(startNumber) // first block to reimburse, use it as the starting time
+		// endTime = startTime + duration
+		endTime := startHeader.Time().Uint64() + reimbursement.Duration.Uint64()
+		parentTime := parentHeader.Time().Uint64()
+		if parentTime >= endTime { // reimburse over
+			return
+		}
+		currentTime := header.Time().Uint64()
+		deltaTime := currentTime - parentTime
```

Comment:
actually don't need to compute time and reimburse based on block time. We just assume the block time is 2 seconds and issue the same fixed number of tokens (hard-coded) for 3 years based on 2-second block time.